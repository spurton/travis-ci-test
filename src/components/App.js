import React, { Component } from 'react';
import { Button } from 'react-bootstrap';

class App extends Component {
  state = {
    gifts: []
  };

  onButtonClick = () => {
    const { gifts } = this.state;

    const ids = gifts.map(gift => gift.id);

    const maxID = gifts.length > 0 ? Math.max(...ids) : 0;

    gifts.push({ id: maxID + 1 });

    this.setState({ gifts });
  };

  render() {
    return (
      <div>
        <h2>Dude</h2>
        <Button className="btn-add" onClick={this.onButtonClick}>
          Click me
        </Button>
        <div id="gift-count">1</div>
      </div>
    );
  }
}

export default App;
