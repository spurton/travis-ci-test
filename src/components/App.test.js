import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

const app = shallow(<App />);

it('renders correctly', () => {
  expect(app).toMatchSnapshot();
})

it('Inializes the `state` with an empty list of gifts', () => {
  expect(app.state().gifts).toEqual([]);
})

it('Adds new gifts to the `state` when you click `add gift`', () => {
  app.find('.btn-add').simulate('click');

  expect(app.state().gifts).toEqual([{ id: 1 }]);
})

it('Displays a current gift count', () => {
  app.find('.btn-add').simulate('click');

  expect(app.find('#gift-count').text()).toEqual("1")
})