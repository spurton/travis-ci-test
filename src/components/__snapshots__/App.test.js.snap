// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Dude
</h2>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="btn-add"
          disabled={false}
          onClick={[Function]}
>
          Click me
</Button>,
        <div
          id="gift-count"
>
          1
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Dude",
        },
        "ref": null,
        "rendered": "Dude",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Click me",
          "className": "btn-add",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Click me",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "1",
          "id": "gift-count",
        },
        "ref": null,
        "rendered": "1",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Dude
</h2>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            className="btn-add"
            disabled={false}
            onClick={[Function]}
>
            Click me
</Button>,
          <div
            id="gift-count"
>
            1
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Dude",
          },
          "ref": null,
          "rendered": "Dude",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Click me",
            "className": "btn-add",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Click me",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1",
            "id": "gift-count",
          },
          "ref": null,
          "rendered": "1",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
